image/config.go

```diff
@@ -68,6 +73,30 @@ func findConfig(w walker, d *descriptor) (*config, error) {
 	}
 }
 
+func checkPlatform(c config) error {
```

Comment:
The desired location for this logic is still up in the air (as far as I can tell), with some folks arguing for just JSON Schema validation in image-spec ([me][1] and [maybe @vbatts][2]) and some (maybe [@philips][3] and [@stevvooe][4]) arguing for as much JSON validation as you can get (JSON Schema or otherwise) without needing to look outside the given JSON blob (e.g.  at children it references).  With the latter approach, this platform check would be landing in the image-spec repo (somewhere under `schema/`?).  It's probably best to wait until @stevvooe has filed his [proposed package structure][5] before moving ahead with this..

[1]: https://github.com/opencontainers/image-spec/pull/337
[2]: http://ircbot.wl.linuxfoundation.org/meetings/opencontainers/2016/opencontainers.2016-10-12-21.01.log.html#l-57
[3]: http://ircbot.wl.linuxfoundation.org/meetings/opencontainers/2016/opencontainers.2016-10-12-21.01.log.html#l-71
[4]: http://ircbot.wl.linuxfoundation.org/meetings/opencontainers/2016/opencontainers.2016-10-12-21.01.log.html#l-74
[5]: http://ircbot.wl.linuxfoundation.org/meetings/opencontainers/2016/opencontainers.2016-10-12-21.01.html